org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/editor/validation/MarkerCreator.java

```diff
@@ -36,28 +39,32 @@ public void createMarker(Issue issue, IResource resource, String markerType) thr
 	 * @since 2.0
 	 */
 	protected void setMarkerAttributes(Issue issue, IResource resource, IMarker marker) throws CoreException {
+		// Do this in one single setAttributes() call, as each set of an attribute is a workspace operation
+		Map<String, String> attributes = new HashMap<>(16);
+
 		String lineNR = "";
 		if (issue.getLineNumber() != null) {
 			lineNR = "line: " + issue.getLineNumber() + " ";
 		}
-		marker.setAttribute(IMarker.LOCATION, lineNR + resource.getFullPath().toString());
-		marker.setAttribute(Issue.CODE_KEY, issue.getCode());		
-		marker.setAttribute(IMarker.SEVERITY, getSeverity(issue));
-		marker.setAttribute(IMarker.CHAR_START, issue.getOffset());
+		attributes.put(IMarker.LOCATION, lineNR + resource.getFullPath().toString());
+		attributes.put(Issue.CODE_KEY, issue.getCode());		
+		attributes.put(IMarker.SEVERITY, getSeverity(issue));
+		attributes.put(IMarker.CHAR_START, issue.getOffset());
 		if(issue.getOffset() != null && issue.getLength() != null)
-			marker.setAttribute(IMarker.CHAR_END, issue.getOffset()+issue.getLength());
-		marker.setAttribute(IMarker.LINE_NUMBER, issue.getLineNumber());
-		marker.setAttribute(Issue.COLUMN_KEY, issue.getColumn());
-		marker.setAttribute(IMarker.MESSAGE, issue.getMessage());
+			attributes.put(IMarker.CHAR_END, issue.getOffset()+issue.getLength());
+		attributes.put(IMarker.LINE_NUMBER, issue.getLineNumber());
+		attributes.put(Issue.COLUMN_KEY, issue.getColumn());
+		attributes.put(IMarker.MESSAGE, issue.getMessage());
 
 		if (issue.getUriToProblem()!=null) 
-			marker.setAttribute(Issue.URI_KEY, issue.getUriToProblem().toString());
+			attributes.put(Issue.URI_KEY, issue.getUriToProblem().toString());
 		if(issue.getData() != null && issue.getData().length > 0) {
-			marker.setAttribute(Issue.DATA_KEY, Strings.pack(issue.getData()));
+			attributes.put(Issue.DATA_KEY, Strings.pack(issue.getData()));
 		}
 		if (resolutionProvider != null && resolutionProvider.hasResolutionFor(issue.getCode())) {
-			marker.setAttribute(FIXABLE_KEY, true);
+			attributes.put(FIXABLE_KEY, true);
 		}
+		marker.setAttributes(attributes);
```

Comment:
am not sure if we should postpose this until we have a newer platform.
otherwise we need to use reflection here

Reply:
to me it does not look like it is worth (at least not from my point of view), but if you like I can spend a little more time and use reflection. For me, I would close the PR and one can retrieve it later if wanted.

As information, on newer platforms, there is, by the way, even a better API that allows to create the marker with its attributes in one workspace operation, which would be even better. The problem is that I think one use it within this classes because they offer a protected method to "set the attributes" and not "to create the maker with its attributes".

Reply:
I think `#setAttributes(..)` should already be available in the oldest supported platform, isn't it?

Reply:
I have squashed the commit, just in case, and fixed the commit to pass the eclipse FDN. 
It compiles properly, doesn't that mean that `#setAttributes(..)` is available in the platform you support?

Reply:
https://github.com/eclipse/xtext-eclipse/issues/1634 says no. do you use a different api?

Reply:
I think #1634 was about `#createMarker(.., attributesMap)` whereas we are looking at `setAttributes(attributesMap)`

Reply:
but is there any benefit to change here?